storage_controller/src/peer_client.rs

```diff
@@ -0,0 +1,107 @@
+use crate::tenant_shard::ObservedState;
+use pageserver_api::shard::TenantShardId;
+use serde::{Deserialize, Serialize};
+use std::collections::HashMap;
+use tokio_util::sync::CancellationToken;
+
+use reqwest::{StatusCode, Url};
+use utils::{backoff, http::error::HttpErrorBody};
+
+#[derive(Debug, Clone)]
+pub(crate) struct PeerClient {
+    hostname: String,
```

Comment:
nit: for configuring http APIs clients, prefer `Uri` to a raw string+int

Reply:
Done in https://github.com/neondatabase/neon/pull/8588/commits/6f0253ada7d52402f8ea04f64b7e9abc1928113b and https://github.com/neondatabase/neon/pull/8588/commits/2a16848c94125307b63bfaf5df95b11aad9f056d